src/main/java/jenkins/plugins/purgejobhistory/PurgeJobHistory.java

```diff
@@ -85,9 +107,12 @@ public static void purge(Job<?, ?> job, boolean resetNextBuildNumber) throws IOE
                 acl.checkPermission(Run.DELETE);
                 lastACL = acl;
             }
+            if (!force && run.isKeepLog()) {
+                continue;
+            }
             run.delete();
         }
-        if (resetNextBuildNumber) {
+        if (resetNextBuildNumber && job.getLastBuild() == null) {
             job.updateNextBuildNumber(1);
```

Comment:
Well I was going for doing nothing if there are any builds left as that is the only way we can ensure that the next number is 1 when the user asked to reset it to 1